k8s-daemonset/README.md

```diff
@@ -24,42 +23,96 @@ docker push <helloworld image name>
 
 ## Deploying
 
-Deploy the hello and world python services, linkerd daemonset, and linkerd-viz
-to the default Kubernetes namespace.
+### Hello World
+
+Start by deploying the hello and the world apps by running:
 
 ```bash
-kubectl apply -f k8s/linkerd.yml
-kubectl apply -f k8s/linkerd-viz.yml
 kubectl apply -f k8s/hello-world.yml
 ```
 
-If you are running on Kubernetes 1.4 or later, you can alternatively run:
+Alternatively, if you are running on Kubernetes 1.4 or later, you can take
+advantage of some newer Kubernetes features, and run:
+
+```bash
+kubectl apply -f k8s/hello-world-1_4.yml
+```
+
+Either of these commands will create new Kubernetes Services and
+ReplicationControllers for both the hello app and the world app.
+
+### linkerd
+
+Next deploy linkerd, to route requests from the hello app to the world app.
+
+For the most basic linkerd DaemonSets configuration, you can run:
 
 ```bash
 kubectl apply -f k8s/linkerd.yml
+```
+
+That command will create linkerd's config file as a Kubernetes ConfigMap, and it
+will use that config file to start linkerd as part of a Kubernetes DaemonSet. It
+will also add a Kuberentes Service for granting external access to linkerd.
+
+For a linkerd configuration that adds TLS to all service-to-service calls, run:
+
+```bash
+kubectl apply -f k8s/linkerd-tls.yml
+```
+
+Or finally, to run linkerd with routing resolutions handled by namerd, run:
+
+```bash
+kubectl apply -f k8s/namerd.yml
+kubectl apply -f k8s/linkerd-namerd.yml
+```
+
+That command will start namerd in your cluster, and configure linkerd to use
+namerd when applying routing rules.
+
+### linkerd-viz
+
+And lastly, deploy linkerd-viz by running:
+
+```bash
 kubectl apply -f k8s/linkerd-viz.yml
```

Comment:
realizing this isn't exactly part of your change, but consider deleting linkerd-viz.yml in this repo and referencing the linkerd-viz repo. the files are nearly identical, except the one here is slightly out of date.